File: programmers/python/programmers_greedy_2.py
def solution(name):
	alpha = {"A" : 0, "B" : 1, "C" : 2, "D" : 3, "E" : 4, "F" : 5, "G" : 6, "H" : 7, "I" : 8, \
			"J" : 9, "K" : 10, "L" : 11, "M" : 12, "N" : 13, "O" : 12, "P" : 11, "Q" : 10, \
			"R" : 9, "S" : 8, "T" : 7, "U" : 6, "V" : 5, "W" : 4, "X" : 3, "Y" : 2, "Z" : 1}
	answer = 0
	check = [False] * len(name)
	check_num = 0
	pos = 0
	for idx, val in enumerate(name):
		if val == "A":
			check[idx] = True
			check_num += 1
	while True:
		if check[pos] is False:
			answer += alpha[name[pos]]
			check[pos] = True
			check_num += 1
		if check_num == len(name):
			break
		rp = pos
		lp = pos
		i = 1
		while True:
			rp = rp + 1 if rp < len(name) - 1 else 0
			if name[rp] != "A" and check[rp] is False:
				pos = rp
				answer += i
				break
			lp = lp - 1 if lp > 0 else len(name) - 1
			if name[lp] != "A" and check[lp] is False:
				pos = lp
				answer += i
				break
			i += 1
	return answer

File: programmers/cpp/programmers_greedy_2.cpp
#include <string>
#include <vector>

using namespace std;

int num_alpha[26] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1 };

int solution(string name) {
	int answer = 0;
	int cur = 0;
	int num = 0;
	int cnt = 0;
	int cnt_right = 0;
	int cnt_left = 0;
	int* check = new int[name.size()]();

	for (int i = 0; i < name.size(); i++)
	{
		if (name[i] != 'A')
			num++;
	}
	while (1)
	{
		answer += (num_alpha[name[cur] - 65] + cnt_right + cnt_left);
		if (name[cur] != 'A')
		{
			check[cur] = 1;
			cnt++;
		}
		if (num == cnt)
			break;
		int temp_right = cur;
		int temp_left = cur;
		cnt_right = 0;
		cnt_left = 0;
		while (1)
		{
			if (temp_right < name.size() - 1)
				temp_right++;
			else
				temp_right = 0;
			cnt_right++;
			if (check[temp_right] == 0 && name[temp_right] != 'A')
			{
				cur = temp_right;
				cnt_left = 0;
				break;
			}
			if (temp_left > 0)
				temp_left--;
			else
				temp_left = name.size() - 1;
			cnt_left++;
			if (check[temp_left] == 0 && name[temp_left] != 'A')
			{
				cur = temp_left;
				cnt_right = 0;
				break;
			}
		}
	}
	delete[] check;

	return answer;
}